### PR TITLE
Mes 73 showme tellme questions

### PR DESCRIPTION
**Pull request by ataylorBJSS**
- fault reducer refactor
- sass updates for expanded sections
- create vehicle check provider for sub-faults (tellMe, showMe) **note:** This data would ideally be placed in a fault store meta-data object but it was simple/quicker to create a separate provider at this time
- create vehicle check component to handle fault recording logic and displays

### CODE REVIEW

File: src/pages/all-on-one-v2/all-on-one-v2.html [hunk 1]
```diff
@@ -63,16 +63,14 @@
...
 
-        <div class="form-section">
-          <all-on-one-form-sub-element-hold-no-modal section="vehicleCheck" text="Vehicle Check"></all-on-one-form-sub-element-hold-no-modal>
-        </div>
+        <all-on-one-form-vc-button></all-on-one-form-vc-button>
 
-        <div class="form-section" [ngClass]="{'manoeuvre-completed': isEcoComplete() || ecoControlOption.isEnabled || ecoPlanningOption.isEnabled}">
-          <all-on-one-form-sub-element-button #ecoButton section="ECO" text="ECO"></all-on-one-form-sub-element-button>
+        <div class="form-section" [ngClass]="{'action-completed': isEcoComplete() || ecoControlOption.isEnabled || ecoPlanningOption.isEnabled}">
+          <all-on-one-form-sub-element-button #ecoButton section="ECOv1" text="ECO"></all-on-one-form-sub-element-button>
```
Comment (zielinskikamil) [inline src/pages/all-on-one-v2/all-on-one-v2.html]: this v1 suffix is needed?

File: src/providers/vehicle-check/vehicle-check.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  /**
+   * Logic to determine which sub-fault should be used to write to the main vehiclecheck fault in state
+   */
+  overwriteFaultWith() {
+    const { faultType: tellMeFault } = this.tellMe;
+    const { faultType: showMeFault } = this.showMe;
+    if (
+      tellMeFault === undefined ||
```
Comment (zielinskikamil) [inline src/providers/vehicle-check/vehicle-check.ts]: what is this ? :D

File: src/components/all-on-one-form-vc-button/all-on-one-form-vc-button.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+declare const Hammer: any;
+@Component({
+  selector: 'all-on-one-form-vc-button',
+  templateUrl: 'all-on-one-form-vc-button.html'
+})
+export class AllOnOneFormVcButtonComponent {
+  tellMe: IVehicleCheck = {};
+  showMe: IVehicleCheck = {};
+  mc: any;
```
Comment (danielchildsbjss) [inline src/components/all-on-one-form-vc-button/all-on-one-form-vc-button.ts]: mc?

File: src/components/tell-me-modal/tell-me-modal.ts [hunk 4]
```diff
@@ -144,9 +144,9 @@ export class TellMeModalComponent {
     this.viewCtrl.dismiss(null, 'dismiss');
   }
 
-  select(itemId: string) {
-    this.selectedTellMeQuestionId = itemId;
-    setTimeout(() => this.viewCtrl.dismiss(itemId), 300);
+  select(item: object) {
```
Comment (danielchildsbjss) [inline src/components/tell-me-modal/tell-me-modal.ts]: Does this have a type?

File: src/pages/pretest-checks/pretest-checks.ts [hunk 5]
```diff
@@ -26,27 +26,28 @@ export class PretestChecksPage {
     isAutomatic: null
   };
 
-  @select(['faults', 'tellMe']) tellMeState$;
+  @select(['faults', 'vehicleCheck'])
+  vcState$;
```
Comment (danielchildsbjss) [inline src/pages/pretest-checks/pretest-checks.ts]: vc?

File: src/providers/vehicle-check/vehicle-check.ts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+      tellMeFault === showMeFault ||
+      (tellMeFault === 'fault' && (showMeFault === 'serious' || showMeFault === 'dangerous')) ||
+      (tellMeFault === 'serious' && showMeFault === 'dangerous')
+    ) {
+      return this.showMe.faultType;
+    }
+
+    return this.tellMe.faultType;
```
Comment (danielchildsbjss) [inline src/providers/vehicle-check/vehicle-check.ts]: Can we simplify this with early returns, something like:

if (tellMeFault === 'dangerous' || showMeFault === 'dangerous')
  return 'dangerous';

if (tellMeFault === 'serious' || showMeFault === 'serious')
  return 'serious';

if (tellMeFault === 'fault' || showMeFault === 'fault')
  return 'fault';

return null; (?)

File: src/pages/all-on-one-v2/all-on-one-v2.html (hunk 1 above)
Comment (ataylorBJSS) [reply src/pages/all-on-one-v2/all-on-one-v2.html]: no, I'll take it out.

File: src/components/all-on-one-form-vc-button/all-on-one-form-vc-button.ts (hunk 3 above)
Comment (ataylorBJSS) [reply src/components/all-on-one-form-vc-button/all-on-one-form-vc-button.ts]: refers to the Hammer manager for press and hold functionality that has been copied from other components. I'm sure what it should stand for but it's required

File: src/pages/pretest-checks/pretest-checks.ts (hunk 5 above)
Comment (ataylorBJSS) [reply src/pages/pretest-checks/pretest-checks.ts]: short for vehicle check

File: src/providers/vehicle-check/vehicle-check.ts (hunk 6 above)
Comment (ataylorBJSS) [reply src/providers/vehicle-check/vehicle-check.ts]: updated to be more specific and quickest first

**Pull request merged**